File: src/ddd/simple/service/courseDesign/CourseDesignService.java
package ddd.simple.service.courseDesign;

import ddd.base.persistence.EntitySet;
import ddd.simple.service.base.BaseServiceInterface;
import ddd.simple.entity.courseDesign.CourseDesign;

public interface CourseDesignService extends BaseServiceInterface {
	public CourseDesign saveCourseDesign(CourseDesign courseDesign) ;
	
	public int deleteCourseDesign(Long courseDesignId) ;
	
	public CourseDesign updateCourseDesign(CourseDesign courseDesign) ;
	
	public CourseDesign findCourseDesignById(Long courseDesignId) ;
	
	public EntitySet<CourseDesign> findCourseDesignByMajorAndSestem(Long majorId,String currentSemester);
	
	public EntitySet<CourseDesign> findAllCourseDesign() ;
 
}

File: WebContent/ddd/simple/view/courseDesign/js/courseDesign/CourseDesignService.js

define(["app"], function (app) {
	app.factory("CourseDesignService", function($resource) {
		
		var courseDesignService = $resource("../DD/CourseDesign/:action", {});
		courseDesignService.saveCourseDesign=function(courseDesign,sucesscb,errorcb)	{
			courseDesignService.save({action:"saveCourseDesign"},{courseDesign:courseDesign},sucesscb,errorcb);
		};
		
		courseDesignService.deleteCourseDesign=function(courseDesignId,sucesscb,errorcb)	{
			courseDesignService.save({action:"deleteCourseDesign"},{courseDesignId:courseDesignId},sucesscb,errorcb);
		};
		
		courseDesignService.updateCourseDesign=function(courseDesign,sucesscb,errorcb) {
			courseDesignService.save({action:"updateCourseDesign"},{courseDesign:courseDesign},sucesscb,errorcb);
		};
		 
		courseDesignService.findCourseDesignById=function(courseDesignId,sucesscb,errorcb) {
			courseDesignService.get({action:"findCourseDesignById",courseDesignId:courseDesignId},sucesscb,errorcb);
		};
		
		courseDesignService.findCourseDesignByMajorAndSestem=function(stuMajor,currentSemester,sucesscb,errorcb) {
			courseDesignService.query({action:"findCourseDesignByMajorAndSestem",stuMajor:stuMajor,currentSemester:currentSemester},sucesscb,errorcb);
		};
		
		courseDesignService.findAllCourseDesign=function(sucesscb,errorcb) {
			courseDesignService.query({action:"findAllCourseDesign"},sucesscb,errorcb);
		}
		return courseDesignService;
})});
